brainiak/isfc.py

```diff
@@ -160,30 +185,48 @@ def isfc(D, collapse_subj=True, return_p=False,
 
     if return_p:
         n_perm = num_perm
+        max_null, min_null = gen_null_array(n_subj, n_perm,
+                                            collapse_subj, float_type)
     else:
         n_perm = 0
 
-    ISFC = np.zeros((n_vox, n_vox, n_subj, n_perm + 1))
+    ISFC = np.zeros((n_vox, n_vox, n_subj), dtype=float_type)
 
-    for p in range(n_perm + 1):
+    for loo_subj in range(D.shape[2]):
+        group = np.mean(D[:, :, np.arange(n_subj) != loo_subj], axis=2)
+        subj = D[:, :, loo_subj]
+        tmp_ISFC = compute_correlation(group, subj).astype(float_type)
+        # Symmetrize matrix
+        tmp_ISFC = (tmp_ISFC+tmp_ISFC.T)/2
+        ISFC[:, :, loo_subj] = tmp_ISFC
+    if collapse_subj:
+        ISFC = np.mean(ISFC, axis=2)
+
+    for p in range(n_perm):
+        # Randomize phases of D to create next null dataset
+        D = phase_randomize(D, random_state)
         # Loop across choice of leave-one-out subject
+        ISFC_mean = np.zeros((n_vox, n_vox), dtype=float_type)

```

Comment:
I would rename this variable to make it clear that this is a null value.

Reply:
Thanks, I renamed it to ISFC_null